http/sys_config_state_test.go

```diff
@@ -9,70 +9,174 @@ import (
 	"testing"
 
 	"github.com/go-test/deep"
+	"github.com/hashicorp/vault/command/server"
+	"github.com/hashicorp/vault/internalshared/configutil"
 	"github.com/hashicorp/vault/vault"
 )
 
 func TestSysConfigState_Sanitized(t *testing.T) {
-	var resp *http.Response
+	cases := []struct {
+		name                    string
+		storageConfig           *server.Storage
+		haStorageConfig         *server.Storage
+		expectedStorageOutput   map[string]interface{}
+		expectedHAStorageOutput map[string]interface{}
+	}{
+		{
+			"raft storage",
```

Comment:
Would it be helpful to add an `inmem` storage test here as well?

Reply:
Sure!

Reply:
See https://github.com/hashicorp/vault/pull/20044/commits/8a9cfb26a947ff86ca27441fea476276dbf9cae7

Reply:
Thanks, @ccapurso!